### PR TITLE
Docker compose config for developing and running tests

### PR DESCRIPTION
**Pull request by dsalahutdinov**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Documentation Update

## Description
Added a couple of features for running project with docker:
 - persistent volumes for database, bundles, node_modules etc
 - support for test environment in docker compose, running specs by `docker-compose run app bundle exec rspec`
 -extracted special base service `app` for running tasks
 - install postgres client to support `rails db` and browsing database 
 - minor documentation improvements

## Added to documentation?

- [ ] docs.dev.to
- [x] readme
- [ ] no documentation needed

### CODE REVIEW

**Comment by benhalpern** [comment]
I don't have a strong opinion here. @maestromac feel free to merge this if you like.

File: docker-compose.yml [hunk 1]
```diff
@@ -1,29 +1,57 @@
...
-      DATABASE_URL: postgres://postgres:mysecretpassword@db:5432/postgres
       YARN_INTEGRITY_ENABLED: "false"
+      RAILS_ENV: ${RAILS_ENV:-development}
+      DATABASE_URL: postgres://postgres:mysecretpassword@db:5432/devto_${RAILS_ENV:-development}
+      DATABASE_CLEANER_ALLOW_REMOTE_DATABASE_URL: "true"
+      WEBPACKER_DEV_SERVER_HOST: webpacker
+
     volumes:
-      - .:/usr/src/app
+      - .:/app
```
[the remark this answers is not in the document]
Comment (dsalahutdinov) [reply docker-compose.yml]: @bibendi great point! fixed

**Comment by dsalahutdinov** [comment]
@maestromac could you please review?

File: Dockerfile [hunk 2]
```diff
@@ -8,24 +13,27 @@ RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources
...
+  GEM_HOME=/bundle \
+  BUNDLE_JOBS=4 \
+  BUNDLE_RETRY=3
+ENV BUNDLE_PATH $GEM_HOME
+ENV BUNDLE_APP_CONFIG=$BUNDLE_PATH \
+  BUNDLE_BIN=$BUNDLE_PATH/bin
+ENV PATH $BUNDLE_BIN:$PATH
 
-ENTRYPOINT ["bundle", "exec"]
+ADD https://github.com/krallin/tini/releases/download/v0.18.0/tini /tini
```
Comment (maestromac) [inline Dockerfile]: Tini's documentation stated that
> NOTE: If you are using Docker 1.13 or greater, Tini is included in Docker itself. This includes all versions of Docker CE. To enable Tini, just pass the --init flag to docker run.

Does that apply here?

**Pull request closed without merging**